packages/admin-ui/src/pages/support/components/UPnP.tsx

```diff
@@ -0,0 +1,165 @@
+import React from "react";
+import { useApi } from "api";
+import { Table } from "react-bootstrap";
+import Loading from "../../../components/Loading";
+import ErrorView from "../../../components/ErrorView";
+import SubTitle from "components/SubTitle";
+import Ok from "components/Ok";
+import Card from "components/Card";
+
+function PortsOpened({ localIp }: { localIp: string }) {
+  const upnpInfo = useApi.getPortsStatus();
+
+  return (
+    <>
+      {upnpInfo.data ? (
+        <>
+          <SubTitle>Opened ports</SubTitle>
+          <Table striped bordered hover>
+            <thead>
+              <tr>
+                <th>Protocol</th>
+                <th>Router port</th>
+                <th>DAppNode port</th>
+                <th>Status</th>
+              </tr>
+            </thead>
+            <tbody>
+              {upnpInfo.data.upnpPortMappings
+                .filter(route => route.ip === localIp) // Check only map routing for the dappnode IP
+                .map(route => {
+                  return (
+                    <tr>
+                      <td>{route.protocol}</td>
+                      <td>{route.exPort}</td>
+                      <td>{route.inPort}</td>
+                      <td>
+                        {upnpInfo.data ? (
+                          upnpInfo.data.portsToOpen.some(
+                            route2 =>
+                              route2.portNumber.toString() === route.exPort // Will display if even having UPNP enabled the port is opened or still closed
+                          ) ? (
+                            <Ok ok={true} msg={"Open"} />
+                          ) : (
+                            <Ok ok={false} msg={"Closed"} />
+                          )
+                        ) : (
+                          <Ok ok={false} msg={"Unknown"} />
+                        )}
+                      </td>
+                    </tr>
+                  );
+                })}{" "}
+            </tbody>
+          </Table>
+        </>
+      ) : upnpInfo.error ? (
+        <ErrorView error={upnpInfo.error} />
+      ) : (
+        <Loading steps={["Loading opened ports"]} />
+      )}
+    </>
+  );
+}
+
+function PortsToOpen() {
```

Comment:
I think you could merge `PortsOpened` and `PortsToOpen` into the same component and reduce code duplication